app/models/video.rb

```diff
@@ -0,0 +1,49 @@
+class Video < ApplicationRecord
+  TWITCH = 'twitch'.freeze
+  YOUTUBE = 'youtube'.freeze
+
+  belongs_to :run
+
+  validates_with VideoValidator
+
+  def twitch?
+    url ? URI.parse(url).host.match?(/^(?:www\.)?(?:twitch\.tv)$/i) : false
+  end
+
+  def youtube?
+    url ? /^(?:https?:\/\/)?(?:www\.)?youtu(?:be\.com|\.be)\/.+/i.match?(url) : false
+  end
+
+  def video_type
+    if twitch?
+      TWITCH
+    elsif youtube?
+      YOUTUBE
+    else
+      nil
+    end
+  end
```

Comment:
`video.video_type` sounds redundant, but I know `type` is reserved, so can we use something like `provider` instead?

Reply:
I like `provider` a lot more than `video_type`.